default.nix

```diff
@@ -45,12 +56,13 @@ rec {
         let
           repoFile = "${dir}/${str}";
           extraFile =
-            "${extraModulesDir}/${pkgs.lib.removePrefix "extra." str}.nix";
+            "${extraModulesDir}/${builtins.replaceStrings [ "." ] [ "/" ] str}.nix";
         in
-        # Load from the extra modules if it starts with extra.
-        if pkgs.lib.hasPrefix "extra." str then import extraFile
-        # Otherwise look in the repo
-        else import repoFile;
+        # First try to import from the user's repository
+        if pkgs.lib.hasPrefix "./" str || pkgs.lib.hasSuffix ".nix" str
```

Comment:
Do you have a strategy how to implement third-party modules later?
I found [source_url](https://github.com/nix-community/nix-direnv#direnv-source_url) quite useful.

Reply:
For now, the user has to use niv of flakes to pull the third-party modules and then use nix to import those modules.

I don't really see a way to handle this in the TOML file directly and be convenient (not having to deal with sha256 manually). 

One idea I have is to extend the module system to add search paths. In that case, the module system gets initialized with a list of paths to look into if the import is a string. It could also deal with importing TOML files natively instead of having that external utility.

Reply:
In theory imports could be also interpreted as flake urls ...

Reply:
Do you mean `imports = ["github:mic92/my-shell-extensions"]` ?